junit-platform-commons/src/main/java/org/junit/platform/commons/util/ModuleUtils.java

```diff
@@ -0,0 +1,77 @@
+/*
+ * Copyright 2015-2017 the original author or authors.
+ *
+ * All rights reserved. This program and the accompanying materials are
+ * made available under the terms of the Eclipse Public License v2.0 which
+ * accompanies this distribution and is available at
+ *
+ * http://www.eclipse.org/legal/epl-v20.html
+ */
+
+package org.junit.platform.commons.util;
+
+import static org.apiguardian.api.API.Status.INTERNAL;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.ServiceLoader;
+import java.util.function.Predicate;
+
+import org.apiguardian.api.API;
+
+/**
+ * Collection of utilities for working with modules.
+ *
+ * <h3>DISCLAIMER</h3>
+ *
+ * <p>These utilities are intended solely for usage within the JUnit framework
+ * itself. <strong>Any usage by external parties is not supported.</strong>
+ * Use at your own risk!
+ *
+ * @since 1.1
+ */
+@API(status = INTERNAL, since = "1.1")
+public final class ModuleUtils {
+
+	/**
+	 * Class finder service providing interface.
+	 */
+	public interface ClassFinder {
```

Comment:
I think a top-level class, e.g. `ModuleClassFinder`, would make it easier to use because it wouldn't require a `$` in the file name of implementors.

Reply:
Mh, thought about that too.

But a) it's an internal SPI and b) you don't specify the `$` when using it in a module descriptor:
```
module org.junit.platform.commons.jpms {
  requires org.junit.platform.commons;

  provides org.junit.platform.commons.util.ModuleUtils.ClassFinder with
      org.junit.platform.commons.jpms.ModuleClassFinder;
}
```